### PR TITLE
Fix logger level defaulting to TRACE in CMake

### PR DESCRIPTION
**Pull request by nurupo**
Apparently we defaulted to TRACE logger level for v0.2.4, v0.2.5, v0.2.6 and v0.2.7 releases when using CMake. We default to INFO logger level when using autotools, since autotools doesn't set any and `logger.h` defaults to INFO if the level is undefined.

There is also this gem which is completely incorrect, logger level won't be set to DEBUG in this case, it will default to whatever `logger.h` defaults to -- INFO.

https://github.com/TokTok/c-toxcore/blob/36f0caaf072128623a4c70e6a614721249bb4c6b/configure.ac#L85

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1212)

### CODE REVIEW

**Comment by nurupo** [comment]
Right, it will fail to compile if user specifies the wrong logger level, but it's clearer if CMake catches that  before compiling and tells what the issue in plain English, rather than deciphering compile error. Also, how often are we planning on modifying the logger level enum anyway for it to be a burden?

```
$ cmake .. -DMIN_LOGGER_LEVEL=FOO
...
CMake Error at CMakeLists.txt:105 (message):
  Unknown value provided for MIN_LOGGER_LEVEL: "FOO", must be one of TRACE,
  DEBUG, INFO, WARNING or ERROR


-- Configuring incomplete, errors occurred!
```

vs

```
$ cmake .. -DMIN_LOGGER_LEVEL=FOO
$ make
Scanning dependencies of target toxcore_static
[  1%] Building C object CMakeFiles/toxcore_static.dir/toxcore/crypto_core.c.o
[  1%] Building C object CMakeFiles/toxcore_static.dir/toxcore/crypto_core_mem.c.o
[  2%] Building C object CMakeFiles/toxcore_static.dir/toxcore/logger.c.o
[  2%] Building C object CMakeFiles/toxcore_static.dir/toxcore/mono_time.c.o
[  3%] Building C object CMakeFiles/toxcore_static.dir/toxcore/network.c.o
./c-toxcore/toxcore/network.c: In function ‘loglogdata’:
<command-line>: error: ‘LOGGER_LEVEL_FOO’ undeclared (first use in this function); did you mean ‘LOGGER_LEVEL_INFO’?
./c-toxcore/toxcore/logger.h:82:22: note: in expansion of macro ‘MIN_LOGGER_LEVEL’
         if (level >= MIN_LOGGER_LEVEL) { \
                      ^~~~~~~~~~~~~~~~
./c-toxcore/toxcore/logger.h:88:34: note: in expansion of macro ‘LOGGER_WRITE’
 #define LOGGER_TRACE(log, ...)   LOGGER_WRITE(log, LOGGER_LEVEL_TRACE  , __VA_ARGS__)
                                  ^~~~~~~~~~~~
./c-toxcore/toxcore/network.c:428:9: note: in expansion of macro ‘LOGGER_TRACE’
         LOGGER_TRACE(log, "[%2u] %s %3u%c %s:%u (%u: %s) | %04x%04x",
         ^~~~~~~~~~~~
<command-line>: note: each undeclared identifier is reported only once for each function it appears in
./c-toxcore/toxcore/logger.h:82:22: note: in expansion of macro ‘MIN_LOGGER_LEVEL’
         if (level >= MIN_LOGGER_LEVEL) { \
                      ^~~~~~~~~~~~~~~~
./c-toxcore/toxcore/logger.h:88:34: note: in expansion of macro ‘LOGGER_WRITE’
 #define LOGGER_TRACE(log, ...)   LOGGER_WRITE(log, LOGGER_LEVEL_TRACE  , __VA_ARGS__)
                                  ^~~~~~~~~~~~
./c-toxcore/toxcore/network.c:428:9: note: in expansion of macro ‘LOGGER_TRACE’
         LOGGER_TRACE(log, "[%2u] %s %3u%c %s:%u (%u: %s) | %04x%04x",
         ^~~~~~~~~~~~
./c-toxcore/toxcore/network.c: In function ‘sendpacket’:
<command-line>: error: ‘LOGGER_LEVEL_FOO’ undeclared (first use in this function); did you mean ‘LOGGER_LEVEL_INFO’?
./c-toxcore/toxcore/logger.h:82:22: note: in expansion of macro ‘MIN_LOGGER_LEVEL’
         if (level >= MIN_LOGGER_LEVEL) { \
                      ^~~~~~~~~~~~~~~~
./c-toxcore/toxcore/logger.h:92:34: note: in expansion of macro ‘LOGGER_WRITE’
 #define LOGGER_ERROR(log, ...)   LOGGER_WRITE(log, LOGGER_LEVEL_ERROR  , __VA_ARGS__)
                                  ^~~~~~~~~~~~
./c-toxcore/toxcore/network.c:477:9: note: in expansion of macro ‘LOGGER_ERROR’
         LOGGER_ERROR(net->log, "attempted to send message of length %u on uninitialised socket", (unsigned)length);
         ^~~~~~~~~~~~
./c-toxcore/toxcore/network.c: In function ‘receivepacket’:
<command-line>: error: ‘LOGGER_LEVEL_FOO’ undeclared (first use in this function); did you mean ‘LOGGER_LEVEL_INFO’?
./c-toxcore/toxcore/logger.h:82:22: note: in expansion of macro ‘MIN_LOGGER_LEVEL’
         if (level >= MIN_LOGGER_LEVEL) { \
                      ^~~~~~~~~~~~~~~~
./c-toxcore/toxcore/logger.h:92:34: note: in expansion of macro ‘LOGGER_WRITE’
 #define LOGGER_ERROR(log, ...)   LOGGER_WRITE(log, LOGGER_LEVEL_ERROR  , __VA_ARGS__)
                                  ^~~~~~~~~~~~
./c-toxcore/toxcore/network.c:558:13: note: in expansion of macro ‘LOGGER_ERROR’
             LOGGER_ERROR(log, "Unexpected error reading from socket: %u, %s", error, strerror);
             ^~~~~~~~~~~~
./c-toxcore/toxcore/network.c: In function ‘networking_poll’:
<command-line>: error: ‘LOGGER_LEVEL_FOO’ undeclared (first use in this function); did you mean ‘LOGGER_LEVEL_INFO’?
./c-toxcore/toxcore/logger.h:82:22: note: in expansion of macro ‘MIN_LOGGER_LEVEL’
         if (level >= MIN_LOGGER_LEVEL) { \
                      ^~~~~~~~~~~~~~~~
./c-toxcore/toxcore/logger.h:91:34: note: in expansion of macro ‘LOGGER_WRITE’
 #define LOGGER_WARNING(log, ...) LOGGER_WRITE(log, LOGGER_LEVEL_WARNING, __VA_ARGS__)
                                  ^~~~~~~~~~~~
./c-toxcore/toxcore/network.c:629:13: note: in expansion of macro ‘LOGGER_WARNING’
             LOGGER_WARNING(net->log, "[%02u] -- Packet has no handler", data[0]);
             ^~~~~~~~~~~~~~
./c-toxcore/toxcore/network.c: In function ‘new_networking_ex’:
<command-line>: error: ‘LOGGER_LEVEL_FOO’ undeclared (first use in this function); did you mean ‘LOGGER_LEVEL_INFO’?
./c-toxcore/toxcore/logger.h:82:22: note: in expansion of macro ‘MIN_LOGGER_LEVEL’
         if (level >= MIN_LOGGER_LEVEL) { \
                      ^~~~~~~~~~~~~~~~
./c-toxcore/toxcore/logger.h:92:34: note: in expansion of macro ‘LOGGER_WRITE’
 #define LOGGER_ERROR(log, ...)   LOGGER_WRITE(log, LOGGER_LEVEL_ERROR  , __VA_ARGS__)
                                  ^~~~~~~~~~~~
./c-toxcore/toxcore/network.c:731:9: note: in expansion of macro ‘LOGGER_ERROR’
         LOGGER_ERROR(log, "invalid address family: %u\n", ip.family.value);
         ^~~~~~~~~~~~
make[2]: *** [CMakeFiles/toxcore_static.dir/build.make:115: CMakeFiles/toxcore_static.dir/toxcore/network.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:909: CMakeFiles/toxcore_static.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
...

**Comment by nurupo** [comment]
There, added a small script that regexes out logger levels and diffs them.

**Comment by iphydf** [comment]
@nurupo lgtm.

**Comment by iphydf** [comment]
Feel free to merge when ready.

**Comment by nurupo** [comment]
Is ready, please merge. Just rebased on master. I can squash first two commits if desired. I'd like to keep the 2nd one and 3rd one separate though.
